### PR TITLE
LLM Batch Workspace Template

### PR DESCRIPTION
**Pull request by brent-anyscale**
Workspace Template for Batch LLM Workloads.

Includes both a Jupyter Notebook walking through using Ray Data with vLLM as well as a Python and job yaml example.

### CODE REVIEW

File: templates/batch-llm/batch_llm.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+
+# Initialize Ray with a Runtime Environment - used to set the Environment Variable `HF_TOKEN`
+# This line can be removed if setting the Environment Variable using a Cluster Environment or in the job yaml file.
+ray.init(runtime_env={"env_vars": {"HF_TOKEN": hf_token}})
+
+input_path = "s3://anonymous@air-example-data/prompts.txt"
+ds = ray.data.read_text(
+    input_path,
+    parallelism=10,
```
Comment (c21) [inline templates/batch-llm/batch_llm.py:22]: Let's remove this line. `parallelism` would be deprecated from Ray Data.

File: templates/batch-llm/batch_llm.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            "prompt": prompt,
+            "generated_text": generated_text,
+        }
+
+
+ds = ds.map_batches(
+    LLMPredictor,
+    concurrency=10,
+    num_gpus=1,
+    batch_size=32,
```
Comment (c21) [inline templates/batch-llm/batch_llm.py:50]: Can we keep the same comments as https://github.com/vllm-project/vllm/pull/2696/files ? The comments would be important for users to understand how to set these parameters.

File: templates/batch-llm/batch_llm.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+)
+
+output_path = os.environ.get("ANYSCALE_ARTIFACT_STORAGE") + "/result"
+ds.write_parquet(output_path)
```
Comment (c21) [inline templates/batch-llm/batch_llm.py:54]: Let's also add the followed same as https://github.com/vllm-project/vllm/pull/2696/files#, and we can comment them out in code. So users can uncomment to try out for a few prompts during testing:

```py
# Peek first 10 results.
# NOTE: This is for local testing and debugging. For production use case,
# one should write full result out as shown below.
# outputs = ds.take(limit=10)
# for output in outputs:
#     prompt = output["prompt"]
#     generated_text = output["generated_text"]
#     print(f"Prompt: {prompt!r}, Generated text: {generated_text!r}")
```

File: templates/batch-llm/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Batch LLM Inference using Anyscale Jobs and vLLM
+
+This template shows you how to:
+* Run a batch LLM inference workload using vLLM and Ray Data
```
Comment (c21) [inline templates/batch-llm/README.md:4]: Let's add a link for vLLM - https://docs.vllm.ai/en/latest/, and Ray Data - https://docs.ray.io/en/latest/data/data.html?

**Review by c21 — COMMENTED** [review]
Thank you @brent-anyscale!

File: templates/batch-llm/README.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# Batch LLM Inference using Anyscale Jobs and vLLM
+
+This template shows you how to:
+* Run a batch LLM inference workload using [vLLM](https://docs.vllm.ai/en/latest) and [Ray Data](https://docs.ray.io/en/latest/data/data.html)
+* Deploy the batch LLM workload as an Anyscale Service
+
+**Note**: This template can be run from this Workspace.
+
+## Review the Jupyter Notebook
+
```
Comment (gvspraveen) [inline templates/batch-llm/README.md:10]: can add a section to run locally in workspace too? 

Basically instruction to run `python batch_llm.py`

Comment (gvspraveen) [reply templates/batch-llm/README.md:10]: oh wait. is that jupyter note book code?

Comment (c21) [reply templates/batch-llm/README.md:10]: Make sense, addressed in https://github.com/anyscale/templates/pull/103 .

**Pull request closed without merging**